Model/Admin/Notification/Messages.php

```diff
@@ -0,0 +1,127 @@
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON
+ * ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ * @author Nosto Solutions Ltd <contact@nosto.com>
+ * @copyright 2017 Nosto Solutions Ltd
+ * @license http://opensource.org/licenses/BSD-3-Clause BSD 3-Clause
+ *
+ */
+
+namespace Nosto\Tagging\Model\Admin\Notification;
+
+use Nosto\Tagging\Helper\Account as NostoHelperAccount;
+use Nosto\Tagging\Helper\Scope as NostoHelperScope;
+use Magento\Framework\Notification\MessageInterface;
+
+class Messages implements MessageInterface
+{
+    private $nostoHelperScope;
+    private $nostoHelperAccount;
+    private $message;
+    private $storeNames;
+    private $display;
+
+    /**
+     * Messages constructor.
+     * @param NostoHelperAccount $nostoHelperAccount
+     * @param NostoHelperScope $nostoHelperScope
+     */
+    public function __construct(
+        NostoHelperAccount $nostoHelperAccount,
+        NostoHelperScope $nostoHelperScope
+    ) {
+        $this->nostoHelperScope = $nostoHelperScope;
+        $this->nostoHelperAccount = $nostoHelperAccount;
+    }
+
+    /**
+     * @return \Magento\Framework\Phrase|string
+     */
+    public function getText()
+    {
+        return __($this->message);
+    }
+
+    /**
+     * @return string
+     */
+    public function getIdentity()
+    {
+        return 'Nosto_Account_Notification';
+    }
+
+    /**
+     * @return bool
+     */
+    public function isDisplayed()
+    {
+        $stores = $this->nostoHelperScope->getStores();
+
+        foreach ($stores as $store) {
+            //Check if the store is connected to Nosto
+            if ($this->nostoHelperAccount->findAccount($store)
+                && $this->nostoHelperAccount->isDomainValid($store) === false) {
+                    $this->storeNames[] = $store->getName();
+                    $this->display = true;
+            }
+        }
+
+        if ($this->display === true) {
+            $this->buildMessage();
+            return true;
+        }
+
+        return false;
+    }
+
+    /**
+     * @return int
+     */
+    public function getSeverity()
+    {
+        // From here you can change notification message type.
+        return MessageInterface::SEVERITY_CRITICAL;
+    }
+
+    /**
+     * Set the value of the message
+     */
+    private function buildMessage()
+    {
+        $message = 'Nosto account is invalid for the stores: ';
```

Comment:
I think we should run these through the translator

Reply:
👍